### PR TITLE
Discovery monitor form improvements

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-2198

### Description Of Changes

Tweaks behavior of discovery monitor form:

Screenshot 2024-06-21 at 16 55 47

* Adds "not scheduled" option to frequency field
* Disables execution start date field if frequency is not selected
* Switches order of frequency and start date fields

### Code Changes

* [ ] Changes to form behavior
* [ ] Remove deprecated ConfigureIntegrationForm component

### Steps to Confirm

* [ ] Configure an integration with valid credentials from the new integrations menu
* [ ] Add a discovery monitor to that integration, then save changes
* [ ] Edit that monitor, then save changes

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx [hunk 1]
```diff
@@ -101,10 +115,17 @@ const ConfigureMonitorForm = ({
 
   const initialValues = {
     name: monitor?.name ?? "",
-    execution_start_date: format(Date.now(), "yyyy-MM-dd'T'hh:mm"),
-    execution_frequency: monitor?.execution_frequency,
+    execution_start_date:
+      monitor?.execution_start_date ?? format(Date.now(), "yyyy-MM-dd'T'HH:mm"),
```
Comment (jpople) [inline clients/admin-ui/src/features/integrations/configure-monitor/ConfigureMonitorForm.tsx:119]: Slight functional change here, note time format changed from `hh:mm` to `HH:mm` to handle AM/PM correctly.

**Comment by eastandwestwind** [comment]
Found a minor issue on my local testing. Steps to repro:

 1. Configure an integration with valid credentials from the new integrations menu. I used BigQuery
 2. Add a discovery monitor to that integration, specifically setting weekly frequency. Note the `Automatic execution start time` field is populated with a date / time.
 3. Save changes
 4. Edit that monitor, see that the `Automatic execution start time` field is NOT populated with a date / time.

Is this expected?

Screenshot 2024-06-24 at 7 02 27 PM

**Comment by jpople** [comment]
@eastandwestwind Thanks for catching that-- got it fixed and added a check in the editing monitor test to verify.

**Review by eastandwestwind — APPROVED** [review]
Confirmed the above issue has been fixed. thanks @jpople !

**Pull request merged**